yaksh/views.py

```diff
@@ -510,27 +561,42 @@ def start(request, questionpaper_id=None, attempt_num=None, course_id=None,
 @login_required
 @email_verified
 def show_question(request, question, paper, error_message=None, notification=None,
-                  course_id=None, module_id=None):
+                  course_id=None, module_id=None, previous_question=None):
     """Show a question if possible."""
     user = request.user
+    quiz = paper.question_paper.quiz
+    quiz_type = 'Exam'
+    can_skip = False
+    if previous_question:
+        delay_time = paper.time_left_on_question(previous_question)
+    else:
+        delay_time = paper.time_left_on_question(question)
+
+    if previous_question and quiz.is_exercise:
+        if delay_time <= 0 or previous_question in paper.questions_answered.all():
+            can_skip = True
+        question = previous_question
     if not question:
         msg = 'Congratulations!  You have successfully completed the quiz.'
         return complete(
             request, msg, paper.attempt_number, paper.question_paper.id,
             course_id=course_id, module_id=module_id
         )
-    if not paper.question_paper.quiz.active:
+    if not quiz.active:
         reason = 'The quiz has been deactivated!'
         return complete(
             request, reason, paper.attempt_number, paper.question_paper.id,
             module_id=module_id
```

Comment:
This will fail once the attempt is started and the quiz is deactivated. You need to send `course_id` as keyword arg to complete function.

Reply:
I have not touched this part!
But since it is an error I will check this, as there are other similar function call!

Reply:
I have checked this is the only function which is missed. But still, you can check